scheds/rust/scx_lavd/src/bpf/main.bpf.c

```diff
@@ -1206,10 +1206,23 @@ static int map_lat_cri_to_lat_prio(u32 lat_cri)
 	return lat_prio;
 }
 
+static u64 calc_starvation_factor(struct task_ctx *taskc)
+{
+	struct sys_stat *stat_cur = get_sys_stat_cur();
+	u64 ratio;
+
+	/*
+	 * Prioritize tasks whose service time is smaller than average.
+	 */
+	ratio = (LAVD_LC_STARVATION_FT * stat_cur->avg_svc_time) /
+		taskc->svc_time;
+	return ratio + 1;
```

Comment:
Shouldn't starvation avoidance be part of the virtual timeline management rather than implemented through boosting interactivity? Is this an artifact of eligible and ineligible timelines being managed separately?

Reply:
Oh that gets removed later. I'm curious why deadline in itself isn't sufficient for starvation avoidance.